vdr/types/interface.go

```diff
@@ -74,6 +74,28 @@ type DocDeactivator interface {
 	Deactivate(id did.DID) error
 }
 
+// DocKeyAdder is the interface that defies functionality to add new keys to a DID document.
+type DocKeyAdder interface {
```

Comment:
This mostly based on the https://golang.org/src/io/io.go which is often referred to when talking about good interface design. As you can see, many single method interfaces and combinations of those. I agree that io.go describes a very low level and generic set of interfaces and here in this DocKeyAdder we are not. Non the less, it gives a level of consistancy. Sometimes it is usefull to only require the dependancy on one function. Especially when the group interfaces are just grouped because they have a common theme and not because they depend on each other. e.g.: you need to resolve a key in auth. Do you inject the complete key store of only a KeyResolver? I would prefer the latter.

Reply:
One big difference: `io` is to be used as lib by other libs. Our stuff is used by us (an app), so why define interfaces that are not used?

Reply:
Removed the interfaces 👍